core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala

```diff
@@ -1,30 +1,42 @@
 package chandu0101.scalajs.react.components
 
-import japgolly.scalajs.react._
 import japgolly.scalajs.react.vdom.html_<^._
 
+import japgolly.scalajs.react.Callback
+import japgolly.scalajs.react.BackendScope
+import japgolly.scalajs.react.ScalaComponent
 import scala.collection.immutable
-import scala.scalajs.js.Date
-import scalacss.ScalaCssReact._
+import scalacss.ProdDefaults._
 
-object ReactTable {
-
-  val cssSettings = scalacss.devOrProdDefaults
-  import cssSettings._
+import scalacss.ScalaCssReact.scalacssStyleaToTagMod
+import chandu0101.scalajs.react.components.ReactSearchBox
+import chandu0101.scalajs.react.components.Pager
+import chandu0101.scalajs.react.components.DefaultSelect
 
-  type Model = Map[String, Any]
+/**
+  * Companion object of ReactTable, with tons of little utilities
+  */
+object ReactTable {
 
   /**
-    *  ._1: String = column name
-    *  ._2: Option[Any => VdomElement] = custom cell
-    *  ._3: Option[(Model,Model) => Boolean] = sorting function
-    *  ._4: Option[Double] = column width interms of flex property
+    * The direction of the sort
     */
-  type Config =
-    (String, Option[Any => VdomElement], Option[(Model, Model) => Boolean], Option[Double])
-
-  val ASC: String = "asc"
-  val DSC: String = "dsc"
+  object SortDirection extends Enumeration {
+    type SortDirection = Value
+    val asc, dsc = Value
+  }
+  /*
+   * Pass this to the ColumnConfig to sort using an ordering
+   */
+  def Sort[T, B](fn: T => B)(implicit ordering: Ordering[B]): (T, T) => Boolean = {
```

Comment:
does this buy us anything over just taking an implicit `Ordering` into the component? `Ordering` has `reverse` for the `asc`/`desc` case